diem-move/diem-framework/DPN/releases/artifacts/current/build/DiemCoreFramework/sources/ParallelExecutionConfig.move

```diff
@@ -0,0 +1,112 @@
+/// This module defines structs and methods to initialize VM configurations,
+/// including different costs of running the VM.
+module DiemFramework::ParallelExecutionConfig {
+    use DiemFramework::DiemConfig::{Self, DiemConfig};
+    use DiemFramework::DiemTimestamp;
+    use DiemFramework::Roles;
+    use Std::Option::{Self, Option};
+
+    /// The struct to hold the read/write set analysis result for the whole Diem Framework.
+    struct ParallelExecutionConfig has copy, drop, store {
+        /// Serialized analysis result for the Diem Framework.
+        /// If this payload is not None, DiemVM will use this config to execute transactions in parallel.
+        read_write_analysis_result: Option<vector<u8>>
+    }
+
+    /// Enable parallel execution functionality of DiemVM by setting the read_write_set analysis result.
+    public fun initialize_parallel_execution(
+        dr_account: &signer,
+    ) {
+        // The permission "UpdateVMConfig" is granted to DiemRoot [[H11]][PERMISSION].
+        Roles::assert_diem_root(dr_account);
+        DiemConfig::publish_new_config(
+            dr_account,
+            ParallelExecutionConfig {
+                read_write_analysis_result: Option::none(),
+            },
+        );
+    }
+
+    public fun set_parallel_execution(
```

Comment:
For parity with `disable_parallel_execution` what do you think of calling this `enable_parallel_execution`?

Reply:
`enable_parallel_execution_with_config`?